doc/source/conf.py

```diff
@@ -368,3 +369,18 @@
 # If true, do not generate a @detailmenu in the "Top" node's menu.
 #
 # texinfo_no_detailmenu = False
+
+
+def setup(app):
+    """ Entry point to sphinx build customisation. """
+    app.connect("autodoc-skip-member", autodoc_skip_member_callback)
+
+
+def autodoc_skip_member_callback(app, what, name, obj, skip, options):
+    """
+    Only skip special methods and functions, including `__init__`, if they have no docstring.
+    """
+    if isinstance(obj, (types.FunctionType, types.MethodType)):
+        return getattr(obj, "__doc__", None) is None
```

Comment:
```suggestion
        if getattr(obj, "__doc__", None) is not None:
            return False  # never skip methods with docstring
```
That

Reply:
alright, alright, i'll add the comment - though never forget how diametrically opposed to the entire concept of comments I am and forever will be

Reply:
also good catch on the logic error there